src/lib/chat/matrix-client.ts

```diff
@@ -168,4 +178,35 @@ export class MatrixClient implements IChatClient {
     createdAt: 0,
     conversationStatus: ConversationStatus.CREATED,
   });
+
+  private mapConversation = (channel): Partial<Channel> => ({
+    id: channel.roomId,
+    name: channel.name,
+    icon: channel.getAvatarUrl(),
+    isChannel: false,
+    isOneOnOne: false,
+    otherMembers: [],
+    lastMessage: null,
+    groupChannelType: GroupChannelType.Public,
+    category: '',
+    unreadCount: 0,
+    hasJoined: true,
+    createdAt: 0,
+    conversationStatus: ConversationStatus.CREATED,
+  });
+
+  private async getFilteredRooms(filterFunc: (roomId: string, dmConversationIds: string[]) => boolean) {
+    if (this.isDisconnected) {
+      return [];
+    }
+
+    if (this.isConnecting) {
+      await this.waitForConnection();
+    }
+
+    const accountData = await this.getAccountData('m.direct');
```

Comment:
Looks like the sdk has enums and functions for many of these things. For example: `EventType.Direct` and `accountData.getContent()`. We should probably use these rather than poking at internal data.

Reply:
Yes for sure, I spotted this in the element code you send me yesterday. Updated to use those!